### ISSUE TITLE
Cinnamon Menu Randomly Changing Sizes

### ISSUE THREAD

**Issue by atiredman** [linuxmint/cinnamon#11712]
Since updating to Cinnamon Version 5.8.2-1 on my Arch Linux laptop, my Cinnamon Menu has been randomly changing sizes. I'm not sure what the trigger is for this behaviour, and it does not always do it, but it is very annoying. I have included a video demonstrating the unwanted behaviour. You can see from the video that this does not occur every time. The first two times I pressed the super key the menu opened at the correct size, however, the third time I pressed it the menu grew larger. Please let me know if you need any further details.

[cinnamon-2023-06-20T005430-0400.webm](https://github.com/linuxmint/cinnamon-menus/assets/32468329/eef8bad2-8d2a-42eb-ab86-bb7818529e76)

**Comment by Soremir** [comment]
Can confirm, have the same issue.

https://github.com/linuxmint/cinnamon/assets/72141678/34003a10-cc0f-4e07-b55b-599ec7d7ef04

**Comment by fredcw** [comment]
Thanks, I've reproduced the issue and looking into it.

**Comment by mtwebster** [comment]
Is this on a Mint theme? Can you share your menu configuration? (right-click on the menu button, Configure, export in the menu)

**Comment by atiredman** [comment]
> Is this on a Mint theme? Can you share your menu configuration? (right-click on the menu button, Configure, export in the menu)

I am using the Mint-Y-Dark-Blue theme. I have attached my menu export. Thank you for looking into this, please let me know if you need any further details.

[Menu.txt](https://github.com/linuxmint/cinnamon/files/11807615/Menu.txt)

**Comment by fredcw** [comment]
@mtwebster It seems the menu.actor size is being reset or invalidated outside of the applet when the menu is closed and reopened. A simple solution is to not set menu.actor size at all and just set this.main_container size instead, as is in fact already being done in _setMenuSize(). The only difference is that the stored menu size will not take into account the padding/border width etc of menu.actor so that if you change the theme, the menu may appear slightly bigger or smaller by a few pixels depending on the theme. I can't seem to find another easy solution.

This can be tested by simply commenting out lines 1434 & 1435 in applet.js which set the menu.actor size.

**Comment by thetredev** [comment]
Sadly I have to reopen the discussion. The issue still persists. System info:
![image](https://github.com/user-attachments/assets/8b6d7a9b-a8c5-4f20-9bb7-7a7cb7ef784d)

**Comment by fredcw** [comment]
@thetredev I can't reproduced this. Is the issue exactly the same as in the video at the top of this post?

**Comment by thetredev** [comment]
@fredcw not sure if it behaves exactly the same. It's just that sometimes the menu shrinks to a very small size, something like 10x10 to give you something to visualize, but I did not measure. And when that happens, diagonal size adjustment via the mouse and the top right corner of the menu goes out of whack: mouse pointer and menu are flickering, size adjustments are not applied - only increasing either width or height on its own works.

I'm sorry I can't give you any more details, but I will make sure to record a video of it when I encounter it again.

One thing to note is that throughout the last 3 years I guess I encountered this issue several times on Arch Linux and Void Linux as well. So, the distro doesn't matter - that I can almost guarantee. But I *did* keep my `$HOME" directory for any of those distro so something in my dotfiles (old config from some old update?) could also be causing problems. I doubt it's related though.
